provider/pkg/await/await.go

```diff
@@ -491,60 +498,147 @@ func ssaUpdate(c *UpdateConfig, liveOldObj *unstructured.Unstructured, client dy
 		if err != nil {
-			return err
+			return fmt.Errorf("unable to parse managed fields from resource %q into fieldpath.Set: %w", c.Inputs.GetName(), err)
 		}
-		patch, err := json.Marshal(patches)
+
+		switch f.Manager {
+		case c.FieldManager:
+			currManagerFieldsSet = currManagerFieldsSet.Union(&s)
+		default:
+			managedFieldsSet = managedFieldsSet.Union(&s)
+		}
+	}
+
+	for _, ignorePath := range c.IgnoreChanges {
+		ipParsed, err := resource.ParsePropertyPath(ignorePath)
 		if err != nil {
-			return err
+			// NB: This shouldn't really happen since we already validated the ignoreChanges paths in the parent Diff function.
+			return fmt.Errorf("unable to parse ignoreField path %q: %w", ignorePath, err)
 		}
-		_, err = client.Patch(c.Context, live.GetName(), types.JSONPatchType, patch, metav1.PatchOptions{})
+
+		pathComponents := strings.Split(ipParsed.String(), ".")
+		pe, err := fieldpath.MakePath(makeInterfaceSlice(pathComponents)...)
 		if err != nil {
-			return err
+			return fmt.Errorf("unable to normalize ignoreField path %q: %w", ignorePath, err)
+		}
+
+		// Drop the field from the inputs if it is present on the cluster and managed by another manager, and is not shared with current manager. This ensures
+		// that we don't get any conflict errors, or mistakenly setting the current field manager as a shared manager of that field.
+		if managedFieldsSet.Has(pe) && !currManagerFieldsSet.Has(pe) {
+			unstructured.RemoveNestedField(c.Inputs.Object, pathComponents...)
+			continue
+		}
+
+		// We didn't find another field manager that is managing this field, so we need to use the last known value applied to
+		// the cluster so we don't unset it or change it to a different value that is not the last known value. This case handles 2 posibilities:
+		//
+		// 1. The field is managed by the current field manager, or is a shared manager, in this case the field needs to be in the request sent to
+		// the server, otherwise it will be unset.
+		// 2. The field is set/exists on the cluster, but for some reason is not listed in the managed fields, in this case we need to set the field to the last
+		// known value applied to the cluster, otherwise it will be unset. This would cause the current field manager to take ownership of the field, but this edge
+		// case probably shouldn't be hit in practice.
+		//
+		// NOTE: If the field has been reverted to its default value, ignoreChanges will still not update this field to what is supplied
+		// by the user in their Pulumi program.
+		lastVal, found, err := unstructured.NestedFieldCopy(liveOldObj.Object, pathComponents...)
+		if found && err == nil {
+			// We only care if the field is found, as not found indicates that the field does not exist in the live state so we don't have to worry about changing the inputs to match
+			// the live state. If this occurs, then Pulumi will set the field back to the declared value. Or should we also ensure that the field is never touch again by Pulumi?
```

Comment:
There's a subtle aspect to ignorechanges that I did not realize until I re-read the docs, which say:
> One reason you would use the ignoreChanges option is to ignore changes in properties that lead to diffs. Another reason is to **change the defaults for a property** without forcing all existing deployed stacks to update or replace the affected resource.

I suppose this means that the inputs should be ignored even if we are the field manager. Does that help answer this question, I dunno.

Reply:
Yes, that's the logic I have defined as well. ignoreChanges always ignores what's declared in the Pulumi program unless that field is not present in the actual live resource/state, then we use the declared value.